lib/tests/streamlit/runtime/scriptrunner/script_runner_test.py

```diff
@@ -276,6 +279,48 @@ def test_run_script(self, filename, text):
             (" ScriptRunner should set the __main__.__file__" "attribute correctly"),
         )
 
+    @patch("streamlit.exception")
+    def test_run_nonexistent_fragment(self, patched_st_exception):
+        """Tests that we raise an exception when trying to run a nonexistent fragment."""
+        scriptrunner = TestScriptRunner("good_script.py")
+        scriptrunner.request_rerun(RerunData(fragment_id="nonexistent_fragment"))
+        scriptrunner.start()
+        scriptrunner.join()
+
+        self._assert_events(
+            scriptrunner,
+            [
+                ScriptRunnerEvent.SCRIPT_STARTED,
+                ScriptRunnerEvent.FRAGMENT_STOPPED_WITH_SUCCESS,
+                ScriptRunnerEvent.SHUTDOWN,
+            ],
+        )
+
+        self._assert_no_exceptions(scriptrunner)
```

Comment:
Based on the test name and description, I would have expected to see a `FRAGMENT_STOPPED_WITH_ERROR` event and an assert that an exception was raised. In case this^ is expected, could you add a comment explaining why we see a success event etc.?

Reply:
Ah, yeah -- let me add an explanatory comment as to why this happens. The reason is that the only type of error that we treat specially is if a user script doesn't compile at all. Other errors are displayed in an `st.exception` element.